utils/zx7/Makefile

```diff
@@ -0,0 +1,112 @@
+#---------------------------------------------------------------------------------
+
+CFLAGS	= -Os \
+		  -Wall \
+		  -funroll-loops \
+		  -fno-trapping-math \
+		  -fno-trapv \
+		  -Wno-switch \
+		  $(MACHDEP) $(INCLUDE) $(DEFINES)
+
+CXXFLAGS =	$(CFLAGS) \
+		  -fpermissive \
+		  -fno-rtti \
+		  -fno-exceptions \
+		  -fno-threadsafe-statics \
+		  -fno-use-cxa-get-exception-ptr 
+
+ASFLAGS	=	$(CFLAGS) 
+
+LDFLAGS	= $(MACHDEP) -O2 -T$(FXCGSDK)/toolchain/prizm.x -Wl,-static -Wl,-gc-sections
+
+#---------------------------------------------------------------------------------
+# list of directories containing libraries, this must be the top level containing
+# include and lib
+#---------------------------------------------------------------------------------
+LIBDIRS	:=
+
+ifneq ($(BUILD),$(notdir $(CURDIR)))
+
+export OUTPUT	:=	$(CURDIR)/../lib/lib$(TARGET)
+export DEPSDIR	:=	$(CURDIR)/$(BUILD)
+
+#---------------------------------------------------------------------------------
+# automatically build a list of object files for our project
+#---------------------------------------------------------------------------------
+CFILES		:=	$(foreach dir,$(SOURCES),$(notdir $(wildcard $(dir)/*.c)))
+CPPFILES	:=	$(foreach dir,$(SOURCES),$(notdir $(wildcard $(dir)/*.cpp)))
+SFILES		:=	$(foreach dir,$(SOURCES),$(notdir $(wildcard $(dir)/*.S)))
+
+#---------------------------------------------------------------------------------
+# use CXX for linking C++ projects, CC for standard C
+#---------------------------------------------------------------------------------
+ifeq ($(strip $(CPPFILES)),)
+	export LD	:=	$(CC)
+else
+	export LD	:=	$(CXX)
+endif
+
+#---------------------------------------------------------------------------------
+# build a list of include paths
+#---------------------------------------------------------------------------------
+export INCLUDE	:=	$(foreach dir,$(INCLUDES), -iquote $(CURDIR)/$(dir)) \
+					$(foreach dir,$(LIBDIRS),-I$(dir)/include) \
+					-I$(CURDIR)/$(BUILD) -I$(LIBFXCG_INC) -I$(CURDIR)
+					
+export VPATH	:=	$(foreach dir,$(SOURCES),$(CURDIR)/$(dir))
+					
+export OFILES	:=	$(addsuffix .o,$(BINFILES)) \
+					$(CPPFILES:.cpp=.o) $(CFILES:.c=.o) \
+					$(SFILES:.S=.o)
+.PHONY: $(BUILD) clean
+
+#---------------------------------------------------------------------------------
+$(BUILD):
+	@[ -d $@ ] || mkdir $@
+	@make --no-print-directory -C $(BUILD) -f $(CURDIR)/Makefile
+
+#---------------------------------------------------------------------------------
+export CYGWIN := nodosfilewarning
+clean:
+	$(RM) -fr $(BUILD) $(OUTPUT).a $(OFILES)
+
+#---------------------------------------------------------------------------------
+else
+
+DEPENDS	:=	$(OFILES:.o=.d)
+
+#---------------------------------------------------------------------------------
+# main targets
+#---------------------------------------------------------------------------------
+$(OUTPUT).a: $(OFILES)
```

Comment:
Sure!
I've built a tool (in utils/zx7/CLI) which can compress and decompress files with zx7.

Reply:
That's a start, but doesn't address the concept of having actual integration with the tools. Right now there's really no reason to offer this library with the toolchain versus encouraging users to include the decompressor in their project, because this doesn't solve any of the hard problems involved in packaging compressed resources into a binary.

What I really want this to be able to do is for the user to load some rules we provide, and then the library handles compressing specified resource files, embedding the data in the output binary, and possibly generating a header file describing the compressed resources.